### PR TITLE
Split Assert class per group of feature

### PR DESCRIPTION
**Pull request by engyebrahim**

### CODE REVIEW

File: src/TestFramework/MSTest.Core/Assertions/Assert.cs [hunk 1]
```diff
@@ -2946,7 +177,7 @@ internal static string ReplaceNulls(object input)
 
     private static int CompareInternal(string expected, string actual, bool ignoreCase, CultureInfo culture)
     {
-        return string.Compare(expected, actual, ignoreCase, culture);
+        return string.Compare(expected, actual, ignoreCase ,culture);
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.cs]: Please revert this change.

File: src/TestFramework/MSTest.Core/Assertions/Assert.cs [hunk 2]
```diff
@@ -50,2771 +46,6 @@ public static Assert That
 
     #endregion
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.cs]: You can remove this region and the helper region.

File: src/TestFramework/MSTest.Core/Assertions/Assert.AreEqual.cs [hunk 3]
```diff
@@ -0,0 +1,1592 @@
...
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
+{
+
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.AreEqual.cs]: Nit: Un-needed blank line.

```suggestion
```

File: src/TestFramework/MSTest.Core/Assertions/Assert.Boolean.cs [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
+{
+
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.Boolean.cs]: Nit: Un-needed blank line.

```suggestion
```

File: src/TestFramework/MSTest.Core/Assertions/Assert.Boolean.cs [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.Boolean.cs]: We should probably follow the pattern used for other files (i.e. rename this file as `Assert.IsTrue`).

File: src/TestFramework/MSTest.Core/Assertions/Assert.EqualsAssertion.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
+{
+
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.EqualsAssertion.cs]: Nit: un-needed blank line

```suggestion
```

File: src/TestFramework/MSTest.Core/Assertions/Assert.EqualsAssertion.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    /// Static equals overloads are used for comparing instances of two types for reference
+    /// equality. This method should <b>not</b> be used for comparison of two instances for
+    /// equality. This object will <b>always</b> throw with Assert.Fail. Please use
+    /// Assert.AreEqual and associated overloads in your unit tests.
+    /// </summary>
+    /// <param name="objA"> Object A </param>
+    /// <param name="objB"> Object B </param>
+    /// <returns> False, always. </returns>
+    [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "We want to compare 'object A' with 'object B', so it makes sense to have 'obj' in the parameter name")]
+    public static new bool Equals(object objA, object objB)
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.EqualsAssertion.cs]: I would keep this method in the `Assert.cs` file. This is just to ensure people are not using equality.

File: src/TestFramework/MSTest.Core/Assertions/Assert.Null.cs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.Null.cs]: File should be named `Assert.IsNull` to follow the pattern of other files.

File: src/TestFramework/MSTest.Core/Assertions/Assert.Type.cs [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+/// <summary>
+/// A collection of helper classes to test various conditions within
+/// unit tests. If the condition being tested is not met, an exception
+/// is thrown.
+/// </summary>
+public sealed partial class Assert
```
Comment (Evangelink) [inline src/TestFramework/MSTest.Core/Assertions/Assert.Type.cs]: File should be named `Assert.IsInstanceOfType` to follow the other files.

**Review by Evangelink — CHANGES_REQUESTED** [review]
Let's not add the new overloads in this PR.

File: src/TestFramework/MSTest.Core/Assertions/Assert.IsNull .cs [hunk 10]
```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (Haplois) [inline src/TestFramework/MSTest.Core/Assertions/Assert.IsNull .cs]: Remove the space just before the extension from the file name.

**Review by Haplois — CHANGES_REQUESTED** [review]

File: test/UnitTests/MSTest.Core.Unit.Tests/Assertions/AssertTests.AreEqualTests.cs [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+
+using Assert = FrameworkV1::Microsoft.VisualStudio.TestTools.UnitTesting.Assert;
+using StringAssert = FrameworkV1::Microsoft.VisualStudio.TestTools.UnitTesting.StringAssert;
+using TestClass = FrameworkV1::Microsoft.VisualStudio.TestTools.UnitTesting.TestClassAttribute;
+using TestFrameworkV2 = FrameworkV2.Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestMethod = FrameworkV1::Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute;
+
+public partial class AssertTests
+{
+    #region AreNotEqual tests.
```
Comment (Evangelink) [inline test/UnitTests/MSTest.Core.Unit.Tests/Assertions/AssertTests.AreEqualTests.cs]: You can remove the region.

**Review by Evangelink — COMMENTED** [review]
One last round of changes and we are good to go! Thanks @engyebrahim

**Review by Evangelink — APPROVED** [review]

**Pull request merged**